### ISSUE TITLE
Reports violations even allowed by settings

### ISSUE THREAD

**Issue by hfrmobile** [DavidAnson/vscode-markdownlint#433]
.markdownlint.json

```json
{
  "MD033": {
    "allowed_elements": ["u", "a", "img", "span", "details", "abbr"]
  },
  "MD013": {
    "line_length": 150,
    "ignore_code_blocks": true,
    "ignore_tables": true
  }
}
```

_markdownlint_ respects that. Git Bash and PowerShell as well.

But **vscode-markdownlint** reports e.g. `span`

**Comment by DavidAnson** [comment]
This issue does not reproduce for me: https://github.com/DavidAnson/playground/tree/vscode-markdownlint-issue-433

Please point to a repository I can use to reproduce the issue you report or explain how to modify my example.

FYI, `ignore_code_blocks` and `ignore_tables` are not valid parameters for MD013: https://github.com/DavidAnson/markdownlint/blob/main/doc/md013.md

**Comment by hfrmobile** [comment]
Thanks for your prompt reply and your effort trying to reproduce it!

Regarding to `ignore_code_blocks` and `ignore_tables`: AI made a suggestion/"improvement" and silly human accepted it...

Maybe it is worth to mention that I'm using Windows 11.

Examples:

Left side you see the **.markdownlint.json** and on the right side some Markdown showing the behavior of some rules.

<img width="1698" height="1025" alt="Image" src="https://github.com/user-attachments/assets/ce7f0bee-5ed5-4587-8d24-f3feec90a3d3" />

1. Markdownlint.ps1 and Extension correctly reporting `table`, `tr`, `td` HTML tags. Both are correct.
2. Markdownlint.ps1 does not report `span` but Extension does. Extension is wrong, shouldn't report it.
3. Markdownlint.ps1 reports line length but Extension doesn't. Extension is wrong, should report it.

Table and code block: both not reporting it, which is correct behavior.

**Comment by DavidAnson** [comment]
You can try my example and confirm that it works for you on Windows. What I'm guessing is happening is that you have the configuration file one directory higher than the project you have opened in VS Code. The extension looks for configuration within the project and will not look in parent directories. I'm guessing whatever script you're using is either configured to use the particular configuration file you want or else is being run from the parent directory. You can configure the extension to use a specific configuration file anywhere on disk, but I would recommend moving the configuration into the project because that's the most portable solution that will work for other people who also work on the project with you.

**Comment by hfrmobile** [comment]
Your guess seems to be 100% correct!

I tried to try your example. Cloned it but didn't get `.markdownlint.json`:

```shell
User@XY MINGW64 /c/Development/ext/DavidAnson/playground (main)
$ ls -la
total 11
drwxr-xr-x 1 User 197121  0 May 18 21:17 ./
drwxr-xr-x 1 User 197121  0 May 18 21:17 ../
drwxr-xr-x 1 User 197121  0 May 18 21:18 .git/
-rw-r--r-- 1 User 197121 14 May 18 21:17 .gitignore
-rw-r--r-- 1 User 197121 41 May 18 21:17 .npmrc
-rw-r--r-- 1 User 197121 63 May 18 21:17 README.md
```

total 11(?)

```shell
User@XY MINGW64 /c/Development/ext/DavidAnson/playground (main)
$ ls -la
total 12
drwxr-xr-x 1 User 197121   0 May 18 21:25 ./
drwxr-xr-x 1 User 197121   0 May 18 21:17 ../
drwxr-xr-x 1 User 197121   0 May 18 21:22 .git/
-rw-r--r-- 1 User 197121  14 May 18 21:17 .gitignore
-rw-r--r-- 1 User 197121 131 May 18 21:26 .markdownlint.json
-rw-r--r-- 1 User 197121  41 May 18 21:17 .npmrc
-rw-r--r-- 1 User 197121 290 May 18 21:28 README.md
```

Downloaded .markdownlint.json manually, then:
total 12 (...)

...that's some weird Git / Git Bash behavior ...

What I can confirm: The Extension takes the config from the current directory, the CLI from user profile. So they behave differently.... Since I am using just one ".markdownlint.json" for all projects/everywhere across projects I'd like to avoid to copy that config file to all projects because when there is a change, I have to adjust copy  it again to all projects + using Markdownfiles not just for (developer) projects also using it outside "projects" and for "normal documents" (instead of Word documents, I'm using Markdown) and Markdown files are spread overall my local disc (synced to cloud) and it would not be practicable to have for each single Markdown file the ".markdownlint.json" as well. We're talking about hundreds of Markdown files.

> whatever script you're using

That confuses me. The markdownlint CLI is developed by you as well, right? My configuration allows me to run markdownlint everywhere. I installed markdownlint long time ago via npm. Later I discovered the VS Code Extension.

According to the [Changelog](https://github.com/DavidAnson/markdownlint/blob/main/CHANGELOG.md) latest version is 0.40.0 but:

```shell
PS> markdownlint --version
0.46.0
```

I checked all the branches and didn't see a newer version than 0.40.0 ...

> You can configure the extension to use a specific configuration file anywhere on disk

That is exactly what I need (see my "special usage" above) but I didn't figure out where I can configure it ...
Can you put me into the right direction? Many thanks in advance!

**Comment by DavidAnson** [comment]
> I tried to try your example. Cloned it but didn't get `.markdownlint.json`:

You cloned main branch and have not yet checked out the branch I link to.

> > whatever script you're using
> 
> That confuses me. The markdownlint CLI is developed by you as well, right? 

In an earlier comment you reference `Markdownlint.ps1` which is not part of either CLI.

> > You can configure the extension to use a specific configuration file anywhere on disk
> 
> That is exactly what I need (see my "special usage" above) but I didn't figure out where I can configure it ...

https://github.com/DavidAnson/vscode-markdownlint#markdownlintconfigfile

**Comment by hfrmobile** [comment]
Using this solved the "configuration issues":

```json
"markdownlint.configFile": "~/.markdownlint.json"
```

`~` expands to the user profile directory where my .markdownlint.json resides.

`Markdownlint.ps1` is just a wrapper that calls your `markdownlint` (_I assumed that this was also developed by you_). It enables the usage of your markdownlint with PS (PowerShell).

Thanks a lot for your effort and clarifications!
Greetings from Austria!
